### PR TITLE
Avoid duplicate action registrations

### PR DESCRIPTION
**Pull request by joyceerhl**
Re https://github.com/microsoft/vscode/pull/186572#discussion_r1245921102

### CODE REVIEW

**Review by TylerLeonhardt — DISMISSED** [review]

File: src/vs/workbench/contrib/share/browser/share.contribution.ts [hunk 1]
```diff
@@ -58,16 +58,21 @@ class ShareWorkbenchContribution {
 		}
 		this.configurationService.onDidChangeConfiguration(e => {
 			if (e.affectsConfiguration(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
-				if (this.configurationService.getValue<boolean>(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
+				if (this.configurationService.getValue<boolean>(ShareWorkbenchContribution.SHARE_ENABLED_SETTING) === true && this._disposables === undefined) {
```
Comment (rzhao271) [inline src/vs/workbench/contrib/share/browser/share.contribution.ts:61]: Could the result of the getValue call be saved somewhere first?

**Review by rzhao271 — CHANGES_REQUESTED** [review]
Also, where's `_disposables` instantiated after this change?
Edit: nvm found it in `registerActions()`. The other feedback comment above still applies, though.

File: src/vs/workbench/contrib/share/browser/share.contribution.ts [hunk 2]
```diff
@@ -58,16 +58,22 @@ class ShareWorkbenchContribution {
 		}
 		this.configurationService.onDidChangeConfiguration(e => {
 			if (e.affectsConfiguration(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
-				if (this.configurationService.getValue<boolean>(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
+				const settingValue = this.configurationService.getValue<boolean>(ShareWorkbenchContribution.SHARE_ENABLED_SETTING;
```
Comment (rzhao271) [inline src/vs/workbench/contrib/share/browser/share.contribution.ts:61]: Missing )

**Review by rzhao271 — DISMISSED** [review]

File: src/vs/workbench/contrib/share/browser/share.contribution.ts (hunk 2 above)
Comment (joyceerhl) [reply src/vs/workbench/contrib/share/browser/share.contribution.ts:61]: Ugh 🤡 thank you!

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**